lib/spack/spack/compilers/adaptor.py

```diff
@@ -79,6 +79,31 @@ def implicit_rpaths(self) -> List[str]:
             result.extend(CompilerPropertyDetector(compiler).implicit_rpaths())
         return result
 
+    def _one_compiler(self, for_name) -> spack.spec.Spec:
+        unique_compilers = list(lang.dedupe(self.compilers.values()))
+        if len(unique_compilers) > 1:
+            raise ValueError(
+                f"Property {for_name} is language-specific and the package has"
+                " more than 1 compiler"
+            )
+        return unique_compilers[0]
+
+    @property
+    def opt_flags(self) -> List[str]:
+        compiler_pkg = self._one_compiler("opt_flags").package
+        return getattr(compiler_pkg, "opt_flags", [])
+
+    @property
+    def debug_flags(self) -> List[str]:
+        compiler_pkg = self._one_compiler("debug_flags").package
+        return getattr(compiler_pkg, "debug_flags", [])
+
+    def opt_flags_for_language(self, lang: str) -> List[str]:
+        return getattr(self.compilers[Languages(lang)].package, "opt_flags", [])
+
+    def debug_flags_for_language(self, lang: str) -> List[str]:
+        return getattr(self.compilers[Languages(lang)].package, "debug_flags", [])
```

Comment:
The adaptor is a temporary measure to avoid having to change calls to `.compiler.<attribute>` to something like:
```
self[language].<attribute>
```
immediately. The idea is to give users (and our `builtin` repo) a grace period to adapt to having compilers as dependencies. 

I'm fine adding back `debug_flags` / `opt_flags` if there are many packages using them [^1], but I would not add new API that wasn't previously in `Compiler`.

[^1]: Sometimes ago we briefly discussed these options in a Thu meeting, and decided to not put them in the adaptor, due to the fact that less than 10 packages were using them in `builtin`.